apps/web/src/components/layout/components/HeaderNav.tsx

```diff
@@ -30,32 +31,41 @@ const menuItem = [
 ];
 const headerIconsSettings = { color: colors.B60, width: 30, height: 30 };
 
+const useBootIntercom = (currentUser, currentOrganization) => {
+  const { boot } = useIntercom();
+
+  const [isIntercomEnabled] = useState<boolean>(!!INTERCOM_APP_ID);
+
+  return useCallback(() => {
+    if (isIntercomEnabled) {
+      boot({
+        email: currentUser?.email,
+        name: currentUser?.firstName + ' ' + currentUser?.lastName,
+        createdAt: currentUser?.createdAt,
+        company: {
+          name: currentOrganization?.name,
+          companyId: currentOrganization?._id as string,
+        },
+      });
+    }
+  }, [isIntercomEnabled, boot, currentUser, currentOrganization]);
+};
+
 export function HeaderNav({}: Props) {
   const { currentOrganization, currentUser, logout } = useContext(AuthContext);
+  const bootIntercom = useBootIntercom(currentUser, currentOrganization);
   const { colorScheme, toggleColorScheme } = useMantineColorScheme();
   const { themeStatus } = useLocalThemePreference();
   const dark = colorScheme === 'dark';
   const { addItem } = useContext(SpotlightContext);
 
-  if (INTERCOM_APP_ID) {
-    const { boot } = useIntercom();
-
-    useEffect(() => {
-      if (currentUser && currentOrganization) {
-        boot({
-          email: currentUser?.email,
-          name: currentUser?.firstName + ' ' + currentUser?.lastName,
-          createdAt: currentUser?.createdAt,
-          company: {
-            name: currentOrganization?.name,
-            companyId: currentOrganization?._id as string,
-          },
-        });
-      }
-    }, [currentUser, currentOrganization]);
-  }
+  useEffect(() => {
+    if (bootIntercom && currentUser && currentOrganization) {
+      bootIntercom();
+    }
+  }, [currentUser, currentOrganization, bootIntercom]);
 
-  const themeTitle = () => {
+  const themeTitle = useCallback(() => {
```

Comment:
To avoid unnecessary re-renders as it is a dependency of the `useEffect` implementation below.